Sources/Apollo/GraphQLQueryWatcher.swift

```diff
@@ -66,7 +69,20 @@ public final class GraphQLQueryWatcher<Query: GraphQLQuery>: Cancellable, Apollo
     guard let dependentKeys = dependentKeys else { return }
 
     if !dependentKeys.isDisjoint(with: changedKeys) {
-      fetch(cachePolicy: .returnCacheDataElseFetch)
+      // First, attempt to reload the query from the cache directly, in order not to interrupt any in-flight server-side fetch.
+      store.load(query: query) { [weak self] result in
+        guard let self = self else { return }
+
+        switch result {
+        case .success:
+          self.callbackQueue.async {
```

Comment:
isn't this retain cycle? self.callbackQueue -> self.resultHandler

Reply:
ahhh good catch - we're defending against self being released in the `load` closure but not on the `async` closure - @flobories do you mind updating this?